scripts/deploy-workbook.sh

```diff
@@ -0,0 +1,213 @@
+#!/bin/sh
+
+# Return current date and time with passed log message
+log() {
+    echo "$(date '+%Y-%m-%d %H:%M:%S'): $1" | tee -a log
+}
+
+# Get user input
+prompt() {
+    varname=$1
+    optional=$3
+    read_str=
+    while :
+    do
+        echo -n "$2"
+        read read_str
+        if [ x$read_str = x"" -a x$optional = x"" ]; then
+            continue
+        else
+            break
+        fi
+    done
+    eval "$varname=$read_str"
+}
+
+# Get target subscription from user input
+prompt subscription_id "Enter target Subscription ID: "
+
+# Get target resource group from user input
+prompt resource_group_name "Enter target Resource Group name: "
+
+# If this script doesn't run on CloudShell, it will execute login to Azure
+if [ x$ACC_CLOUD = x"" ]; then
+  # Get tenant from user input
+  prompt tenant "Enter target Tenant name(optional): " this-is-option
+  [ x$tenant != x"" ] && tenant="-t $tenant"
+
+  # az login
+  log "Login to Azure"
+  az login $tenant --use-device-code
+fi
+
+log "Check existance of target resource group"
+az group show --name $resource_group_name --subscription $subscription_id
+if [ $? -ne 0 ]; then
+    log "Please input correct SubscriptionID and Resource Group name"
+    exit 1
+fi
+
+base_url="https://raw.githubusercontent.com/Azure/reliability-workbook/main/"
+# Download file list
+[ ! -f workbook_filelist ] && wget $base_url/artifacts/workbook_filelist
+cat workbook_filelist | while read f
+do
+    log "Download workbook file: $f"
+    if [ -f $f ]; then
+        log "Skip download because the file already exists"
+    else
+        wget $base_url/artifacts/$f
+    fi
+done
+
+# Deploy Workbook
+for f in *.workbook
+do
+  filename_base=`basename $f .workbook`
+  filename=`basename $f`
+  file_size=$(stat -c%s "$f")
+  if [ ! -e ${filename_base}_id -o $file_size -eq 0 ]; then
+    log "Deploy ${filename_base}"
+    { az deployment group create --name $filename_base -g $resource_group_name --template-uri $base_url/artifacts/azuredeploy.json --parameters serializedData=@$filename --parameters name=$filename_base --query 'properties.outputs.resource_id.value' -o json; } > ${filename_base}_id &
+  fi
+done
+
+# Wait for all deployment processes
+wait
+
+[ ! -e workbook.tpl.json ] && wget $base_url/build/templates/workbook.tpl.json
+for f in *_id
+do
+    resource_id=`cat $f | tr -d '\"'`
+    # Get resource type from filename (e.g.: ReliabilityWorkbookExport.workbook -> export)
+    resource_type=`echo $f | sed -e 's/.*Workbook\([^.]*\).*_id/\L\1/g'`
+    # Replace placeholder in the file
+    sed -i "s#\\\${${resource_type}_workbook_resource_id}#$resource_id#g" workbook.tpl.json
```

Comment:
Could you download and run it on Linux environment?
This error seems to occur because of the return code.
If this error also occurs on Linux environment, it's unexpected.

Reply:
Let me try it in WSL and report back

Reply:
@tsubasaxZZZ This is again breaking in WSL. I am executing the script from bash

![image](https://user-images.githubusercontent.com/11145865/235877972-06533213-ad0e-4568-b826-a01d37a02a7e.png)

![image](https://user-images.githubusercontent.com/11145865/235878096-73b675d3-6a9b-4e14-8db8-745575e217f3.png)

Reply:
@infosatheesh2020
The error did not occur in my environment.
Your script seems to be clone git on Windows. And also, if you are using the az command on Windows via interop, it may cause the same problem.
![image](https://user-images.githubusercontent.com/3822284/236450758-e8e892d7-1603-4f70-8f59-3708888df5d4.png)

So, please check the following:
1. Are you cloning or downloading the script on WSL?
```shell
cd ~ # Move home dir(Not Windows filesystem)
wget https://raw.githubusercontent.com/tsubasaxZZZ/reliability-workbook/develop-deployment-tool/scripts/deploy-workbook.sh
sh deploy-workbook.sh
```
2. Are you using the pure Linux az command?
```shell
$ which az
/mnt/c/Program Files (x86)/Microsoft SDKs/Azure/CLI2/wbin/az <---- this is interop ver
```
```shell
$ which az
/usr/bin/az <---- this is not interop ver
```